stellargraph/layer/graph_classification.py

```diff
@@ -189,3 +180,145 @@ def in_out_tensors(self):
+                             activations=["tanh","tanh", "tanh", "tanh"],
+                             generator=generator,
+                )
+            x_inp, x_out = model.in_out_tensors()
+
+            x_out = Conv1D(filters=16, kernel_size=97, strides=97)(x_out)
+            x_out = MaxPool1D(pool_size=2)(x_out)
+            x_out = Conv1D(filters=32, kernel_size=5, strides=1)(x_out)
+            x_out = Dense(units=128, activation="relu")(x_out)
+            x_out = Dropout(rate=0.5)(x_out)
+            predictions = Dense(units=1, activation="sigmoid")(x_out)
+
+            model = Model(inputs=x_inp, outputs=predictions)
+
+
+    Args:
+        layer_sizes (list of int): list of output sizes of the graph GCN layers in the stack.
+        activations (list of str): list of activations applied to each GCN layer's output.
+        k (int): size (number of rows) of output tensor.
+        generator (GraphGenerator): an instance of :class:`GraphGenerator` class constructed on the graphs used for
+            training.
+        bias (bool, optional): toggles an optional bias in graph convolutional layers.
+        dropout (float, optional): dropout rate applied to input features of each GCN layer.
+        kernel_initializer (str or func, optional): The initialiser to use for the weights of each graph
+            convolutional layer.
+        kernel_regularizer (str or func, optional): The regulariser to use for the weights of each graph
+            convolutional layer.
+        kernel_constraint (str or func, optional): The constraint to use for the weights of each layer graph
+            convolutional.
+        bias_initializer (str or func, optional): The initialiser to use for the bias of each layer graph
+            convolutional.
+        bias_regularizer (str or func, optional): The regulariser to use for the bias of each layer graph
+            convolutional.
+        bias_constraint (str or func, optional): The constraint to use for the bias of each layer graph
+            convolutional.
+
+    """
+
+    def __init__(
+        self,
+        layer_sizes,
+        activations,
+        k,
+        generator,
+        bias=True,
+        dropout=0.0,
+        kernel_initializer=None,
+        kernel_regularizer=None,
+        kernel_constraint=None,
+        bias_initializer=None,
+        bias_regularizer=None,
+        bias_constraint=None,
+    ):
+        super().__init__(
+            layer_sizes=layer_sizes,
+            activations=activations,
+            generator=generator,
+            bias=bias,
+            dropout=dropout,
+            kernel_initializer=kernel_initializer,
+            kernel_regularizer=kernel_regularizer,
+            kernel_constraint=kernel_constraint,
+            bias_initializer=bias_initializer,
+            bias_regularizer=bias_regularizer,
+            bias_constraint=bias_constraint,
+        )
+
+        if not isinstance(k, int):
```

Comment:
(Just an FYI, doesn't have to change here)

This validation seems better suited to `SortPooling`, and could be done more simply with `core.validation.require_integer_in_range(k, "k", min_val=1)`:

https://github.com/stellargraph/stellargraph/blob/c5af1bf37bb73cd9b2d5b6d30ce971f0a9331e9d/stellargraph/core/validation.py#L59-L67